packages/network/src/createTxQueue.ts

```diff
@@ -227,14 +227,30 @@ export function createTxQueue<C extends Contracts>(
           error &&
           (("code" in error && error.code === "NONCE_EXPIRED") ||
             JSON.stringify(error).includes("transaction already imported"));
+        const incorrectGasEstimation =
+          error &&
+          (JSON.stringify(error).includes("gas required exceeds allowance") ||
+            JSON.stringify(error).includes("intrinsic gas too low"));
         console.log(
           `[TXQueue] TX Sent (error=${!!error}, isMutationError=${!!isNonViewTransaction} incNonce=${!!shouldIncreaseNonce} resetNonce=${!!shouldResetNonce})`
         );
-        // Nonce handeling
-        if (shouldIncreaseNonce) incNonce();
+
         if (shouldResetNonce) await resetNonce();
-        // Bubble up error
-        if (error) txRequest.cancel(error);
+        const canRetry = incorrectGasEstimation || shouldResetNonce;
+
+        if (canRetry) {
+          queue.add(uuid(), {
+            execute: txRequest.execute,
+            estimateGas: txRequest.estimateGas,
+            cancel: txRequest.cancel,
+            stateMutability,
+          });
```

Comment:
we should probably count the number of retries so this doesn't end up infinitely looping because of bad gas estimate

Reply:
ya fair

Reply:
if txs are queued in order, what happens to all the txs after this one in the queue if this one is retried?

Reply:
yeah this would all collapse if you queued a bunch of txs that depend on each other

Reply:
but i believe only the first one should retry once, then everything will properly fail in confirmation and not retry